File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.js
/**
 * @param {string[]} strs
 * @return {number}
 */
var minDeletionSize = function(strs) {
    let del = 0;
    const rows = strs.length;
    const cols = strs[0].length;

    for (let c = 0; c < cols; c++) {
        let prev = strs[0].charCodeAt(c);

        for (let r = 1; r < rows; r++) {
            let curr = strs[r].charCodeAt(c);
            if (curr < prev) {
                del++;
                break;
            }
            prev = curr;
        }
    }
    return del;
};

File: 0944-delete-columns-to-make-sorted/0944-delete-columns-to-make-sorted.cs
public class Solution {
    public int MinDeletionSize(string[] strs) {
        int del = 0;
    int rows = strs.Length;
    int cols = strs[0].Length;

    for (int c = 0; c < cols; c++)
    {
        int prev = strs[0][c];  

        for (int r = 1; r < rows; r++)
        {
            int curr = strs[r][c];

            if (curr < prev)
            {
                del++;
                break;
            }

            prev = curr;
        }
    }

    return del;
    }
}